### ISSUE TITLE
[Feature]: Preserve or configure line endings when formatting files

### ISSUE THREAD

**Issue by Gravifer** [Enter-tainer/typstyle#496]
### Description

Typstyle currently normalizes formatted output to LF line endings. As a result, a Typst file that is otherwise fully formatted but uses consistent CRLF line endings is reported by `typstyle --check` as requiring reformatting.

This is inconvenient for repositories that intentionally use CRLF through EditorConfig or Git checkout policy, especially on Windows.

### Minimal reproduction

Create an otherwise canonically formatted Typst file with CRLF line endings. 
Then, using Typstyle 0.15.1:

-----

```powershell
[IO.File]::WriteAllBytes(
  "example.typ",
  [Text.Encoding]::UTF8.GetBytes("#let value = 1`r`n")
)

typstyle --check example.typ
$LASTEXITCODE
```

<details>
<summary><i>(the same in other shells)</i></summary>

#### Git Bash

```bash
printf '#let value = 1\r\n' > example.typ

typstyle --check example.typ
echo $?
```

#### Command Prompt

```batch
>example.typ echo #let value = 1

typstyle --check example.typ
echo %ERRORLEVEL%
```
</details>

-----

The command exits with status 1. `typstyle --diff example.typ` reports a change even though the Typst formatting is already canonical and only the line endings differ.

Running `typstyle --inplace example.typ` rewrites the file with LF line endings.

### Current behavior

The formatter produces LF output, and check mode compares that output directly with the original input. Consequently:

- consistently formatted CRLF files fail `--check`;
- `--diff` may display the whole file as changed;
- `--inplace` changes the repository's line-ending convention;
- a repository using `end_of_line = crlf` cannot use `typstyle --check` directly.

### Expected behavior

For file inputs, Typstyle should ideally preserve the file's existing consistent line-ending style, or provide an explicit way to select it.

One possible interface would be:

```text
--line-ending auto | lf | crlf
```

Suggested semantics:

- `auto`: detect and preserve each input file's consistent line ending;
- `lf`: emit LF;
- `crlf`: emit CRLF;
- mixed line endings: either report an error or normalize according to an explicit option;
- stdin/stdout may continue to use LF by default.

In check mode, the formatter output would first be converted to the selected or detected line-ending style before comparison with the input.

The core formatter could continue using LF as its canonical internal representation; detection and restoration could remain a CLI file-I/O concern.

### Alternatives

A weaker alternative would be for `--check` to ignore differences consisting only of LF versus CRLF. However, preserving or configuring line endings would also prevent `--inplace` from unexpectedly changing the repository convention.

### Environment

- Typstyle: 0.15.1
- Platform observed: Windows
- Repository policy: EditorConfig `end_of_line = crlf`, Git `core.autocrlf = true`
